### PR TITLE
[Model Monitor] - Add support for timeseries in metrics outputter

### PR DESCRIPTION
**Pull request by mastloui-msft**

### CODE REVIEW

File: assets/model_monitoring/components/generation_safety_quality/generation_safety_quality_signal_monitor/spec.yaml [hunk 1]
```diff
@@ -138,13 +140,14 @@ jobs:
...
     type: spark
-    component: azureml://registries/azureml/components/model_monitor_metric_outputter/versions/0.3.4
+    component: azureml://registries/azureml/components/model_monitor_metric_outputter/versions/0.3.5
     inputs:
       signal_metrics:
         type: mltable
         path: ${{parent.jobs.compute_metrics.outputs.signal_metrics}}
       signal_type: "GenerationSafetyQuality"
       signal_name: ${{parent.inputs.signal_name}}
+      metric_timestamp: ${{parent.inputs.monitor_current_time}}
```
Comment (chupadhy) [inline assets/model_monitoring/components/generation_safety_quality/generation_safety_quality_signal_monitor/spec.yaml:150]: Missing param - monitor_name here. I think this got trickled down to service also

Comment (chupadhy) [reply assets/model_monitoring/components/generation_safety_quality/generation_safety_quality_signal_monitor/spec.yaml:150]: Actually service is fine

Comment (mastloui-msft) [reply assets/model_monitoring/components/generation_safety_quality/generation_safety_quality_signal_monitor/spec.yaml:152]: Added it, thank you!

File: assets/model_monitoring/components/src/model_monitor_metric_outputter/runmetric_client.py [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+        metrics = {}
+        metrics["value"] = value
+        if threshold is not None:
+            metrics["threshold"] = float(threshold)
+        self.publish_metrics(run_id=run_id, metrics=metrics, step=step)
+
+    def publish_metrics(self, run_id: str, metrics: dict, step: int):
+        """Publish metrics to the run metrics store."""
+        print(
+            f"Publishing metric {metrics} to run {run_id} at step {step}."
```
Comment (yetamsft) [inline assets/model_monitoring/components/src/model_monitor_metric_outputter/runmetric_client.py:154]: nit: could you wrap the metrics dictionary like '{metrics}' for readability?

Comment (mastloui-msft) [reply assets/model_monitoring/components/src/model_monitor_metric_outputter/runmetric_client.py:154]: May I address this as a follow-up once I'm back? Will change whilst doing the samples work

File: assets/model_monitoring/components/src/model_monitor_metric_outputter/builder/metric_output_builder.py [hunk 3]
```diff
@@ -87,12 +101,47 @@ def _add_metric(self, metrics_dict: dict, metric_name: str, metric: dict, group_
                     else:
                         self._create_metric_groups_if_not_exist(cur_dict, group_name)
                         cur_dict[GROUPS][group_name] = metric
+                        cur_dict[GROUPS][group_name][TIMESERIES] = self._create_timeseries(
```
Comment (yetamsft) [inline assets/model_monitoring/components/src/model_monitor_metric_outputter/builder/metric_output_builder.py:104]: I thought "timeseries" metrics are not always generated. Are they?

Comment (mastloui-msft) [reply assets/model_monitoring/components/src/model_monitor_metric_outputter/builder/metric_output_builder.py:104]: They're always generated when a value/threshold is present!

File: assets/model_monitoring/components/src/model_monitor_metric_outputter/run.py [hunk 4]
```diff
@@ -24,16 +27,20 @@ def run():
...
+    arg_parser.add_argument("--metric_timestamp", type=str)
     arg_parser.add_argument("--signal_output", type=str)
 
     args = arg_parser.parse_args()
 
     metrics: List[Row] = read_mltable_in_spark(args.signal_metrics).collect()
 
-    metrics_dict = MetricOutputBuilder(metrics).get_metrics_dict()
+    runmetric_client = RunMetricClient()
+    metrics_dict = MetricOutputBuilder(runmetric_client, args.monitor_name, args.signal_name, metrics) \
```
Comment (yetamsft) [inline assets/model_monitoring/components/src/model_monitor_metric_outputter/run.py:42]: nit: the trailing backslash is not needed when concatenating a dot.

**Review by yetamsft — APPROVED** [review]
Looks good.

**Review by chupadhy — APPROVED** [review]

File: assets/model_monitoring/components/src/model_monitor_metric_outputter/builder/metric_output_builder.py (hunk 3 above)
Comment (chupadhy) [reply assets/model_monitoring/components/src/model_monitor_metric_outputter/builder/metric_output_builder.py:104]: Resolving this comment

File: assets/model_monitoring/components/src/model_monitor_metric_outputter/runmetric_client.py (hunk 2 above)
Comment (chupadhy) [reply assets/model_monitoring/components/src/model_monitor_metric_outputter/runmetric_client.py:154]: Added this WIT - https://msdata.visualstudio.com/Vienna/_workitems/edit/2659503

File: assets/model_monitoring/components/src/model_monitor_metric_outputter/run.py (hunk 4 above)
Comment (chupadhy) [reply assets/model_monitoring/components/src/model_monitor_metric_outputter/run.py:42]: https://msdata.visualstudio.com/Vienna/_workitems/edit/2659503
Added this WIT for fixing this comment

**Pull request merged**